src/Rules/DoctrineKeyValueStyleRule.php

```diff
@@ -126,74 +125,81 @@ public function processNode(Node $callLike, Scope $scope): array
 
         $checkIntegerRanges = QueryReflection::getRuntimeConfiguration()->isParameterTypeValidationStrict();
 
-        // Table name may be escaped with backticks
-        $argTableName = trim($tableType->getValue(), '`');
-        $table = $schemaReflection->getTable($argTableName);
-        if (null === $table) {
-            return [
-                RuleErrorBuilder::message('Query error: Table "' . $argTableName . '" does not exist')->line($callLike->getLine())->build(),
-            ];
-        }
-
-        // All array arguments should have table columns as keys
         $errors = [];
-        foreach ($arrayArgPositions as $arrayArgPosition) {
-            // If the argument doesn't exist, just skip it since we don't want
-            // to error in case it has a default value
-            if (! \array_key_exists($arrayArgPosition, $args)) {
-                continue;
-            }
-
-            $argType = $scope->getType($args[$arrayArgPosition]->value);
-            if (! $argType instanceof ConstantArrayType) {
-                $errors[] = 'Argument #' . $arrayArgPosition . ' is not a constant array, got ' . $argType->describe(VerbosityLevel::precise());
+        foreach ($tableNames as $tableName) {
```

Comment:
I think this logic could benefit from extracting some parts in private methods.

atm its quite un-readable